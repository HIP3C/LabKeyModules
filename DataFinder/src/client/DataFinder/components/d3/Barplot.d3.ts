import * as d3 from 'd3'
import { PlotDatum } from '../../../typings/CubeData';

// ================================================================== //
/* 
This is a barplot component which takes the following arguments in 
the props: 
  * data
  * name (id for the div to stick it in)
  * width
  * height

*/
// Types

interface D3BarplotConfig {
    width: number;
    height: number;
    labels: string[];
    countMetric: string;
    barColor: string;
}


interface D3Barplot {
    create: (id: string, data: PlotDatum[], config: D3BarplotConfig) => void;
    update: (id: string, data: PlotDatum[], config: D3BarplotConfig) => void;
}


const xaxisTitle = {height: 40}

const createBarplot = (id: string, data: PlotDatum[], config: D3BarplotConfig) => {

    // Create margins
    const barplotHeight = config.height - xaxisTitle.height - 10
    const margin = { top: 0, right: 15, bottom: 0, left: 100 },
    width = config.width - margin.left - margin.right,
    height = barplotHeight - margin.top - margin.bottom;

    // Add svg for bars and separate one for x-axis to allow
    const svg = d3
        .select("#barplot-container-" + id).select("svg")
        .attr("height", barplotHeight)
        .attr("width", config.width)
        .attr("id", "svg-barplot-" + id)

    const xAxisSvg = d3.select("#xaxis-" + id).select("svg")
        .attr("width", config.width)
        .attr("height", xaxisTitle.height)


    // Get long and short versions of y-axis labels
    const labels = config.labels.map(l => {
        let short = l
        const maxLength = Math.round( margin.left / 5.5 )
        if (l.length > maxLength) short = l.slice(0,  maxLength - 3) + "..."
        return ({ label: l, shortlabel: short })
    })
    const yaxisScale = d3
        .scaleBand()
        .domain(labels.map(l => l.label))
        .range([height, 0]);

    // Add container for bars
    const barplot = svg
        .append("g")
        .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
        .attr("id", "bars-" + id)
        .attr("style", "overflow:auto")

    // Add y-axis short labels
    const yaxisLabels = svg.append("g")
        .attr("id", "yaxis-labels-short-" + id)
        .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
    yaxisLabels.selectAll("text")
        .data(labels)
        .enter()
        .append("text")
        .attr("x", -5)
        .attr("y", (d) => {
            return yaxisScale((d.label)) + yaxisScale.bandwidth() / 1.5;
        })
        .attr("text-anchor", "end")
        .attr("font-size", ".8em")
        .text(d => d.shortlabel)

    // Insert vertical grid lines (before bars)
    const xaxisGrid = svg.insert("g", ":first-child")
        .attr("id", "xaxis-grid-" + id)
        .attr(
            "transform",
            `translate(${margin.left}, ${(height + 5 + margin.top)})`
        )

    // x-axis label container (scale changes on update)
    const xaxisLabels = xAxisSvg.append("g")
        .attr("class", "x axis")
        .attr(
            "transform",
            `translate(${margin.left}, 0)`
        )

    // x-axis title
    xAxisSvg.append("g")
        .attr(
            "transform",
            `translate(${margin.left}, 0)`
        )
        .append("text")
        .attr("x", (width) / 2)
        .attr("y", 30)
        .attr("class", "x-axis-title")
        .text(config.countMetric == "studyCount" ? "Studies" : "Participants")
        .attr("text-anchor", "middle")
        .attr("font-size", "0.8em")

    // y-axis long labels container (for hover)
    const yaxisLongLabels = svg.append("g")
        .attr("id", "yaxis-labels-" + id)
        .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
}

const updateBarplot = (id: string, data: PlotDatum[], config: D3BarplotConfig) => {
    // Prelims -----------------
    const svg = d3.select("#barplot-container-" + id).select("svg")
    const xAxisSvg = d3.select("#xaxis-" + id).select("svg")
    const barplot = d3.select("#bars-" + id)
    const xaxisGrid = svg.select("#xaxis-grid-" + id)
    const xaxisLabels = xAxisSvg.selectAll(".x.axis")
    // yaxisLongLabels -- a group on top of the othe
    const yaxisLongLabels = svg.select("#yaxis-labels-" + id) 

    

    // Create margins
    const barplotHeight = config.height - xaxisTitle.height - 10
    const margin = { top: 0, right: 15, bottom: 0, left: 100 },
    width = config.width - margin.left - margin.right,
    height = barplotHeight - margin.top - margin.bottom;
    
    const yaxisScale = d3
        .scaleBand()
        .domain(config.labels.map(l => l))
        .range([height, 0]);
    

    // Make the updates ------------
    // Bind data
    // Get range of data for x-axis
    const dataRange = [0, 10];
    const countMetric = config.countMetric
    data.forEach((v) => (v[countMetric] > dataRange[1]) && (dataRange[1] = v[countMetric]))
    const xaxisScale = d3
        .scaleLinear()
        .domain(dataRange).nice()
        .range([0, width]);
    const updateXaxis: any = d3.axisBottom(xaxisScale).ticks(3).tickSize(0)
    const updateXaxisGrid: any = d3.axisBottom(xaxisScale).ticks(3).tickSize(-(height + 5)).tickFormat(() => "")

    // Bind the data
    const boxes = barplot.selectAll("rect").data(data);

    const yaxisHoverLabels = yaxisLongLabels
        .selectAll(".yaxis-long-label-container")
        .data(data)

    // ---- enter ----
    // bars on the barplot
    boxes
        .enter()
        .append("rect")
        .attr("class", "rect")
        .attr("x", xaxisScale(0))
        .attr("width", function (d: PlotDatum) {
            return xaxisScale(d[countMetric]);
        })
        .attr("y", function (d: PlotDatum) {
            return yaxisScale(d.member);
        })
        .attr("height", yaxisScale.bandwidth() - 1)
        .style("fill", config.barColor)
    
    // Labels for hover
    const labelContainers = yaxisHoverLabels
        .enter()
        .append("g")
        .attr("class", "yaxis-long-label-container") 

    labelContainers
        .append("text")
        .attr("class", "label-number")
        .attr("x", config.width - margin.left)
        .attr("y", d => yaxisScale((d.member)) + yaxisScale.bandwidth() / 1.5)
        .attr("text-anchor", "end")
        .text(d => d[countMetric] == null ? 0 : d[countMetric])
        .attr("font-size", ".8em")
        .attr("fill", "transparent")
        
    // Rect over the whole area around the bar to use for adding mouseover event
    labelContainers
        .append("rect")
        .attr("class", "big-rect-cover")
        .attr("x", - margin.left)
        .attr("y", (d: PlotDatum) => {return(yaxisScale(d.member))})
        .attr("width", config.width)
        .attr("height", yaxisScale.bandwidth())
        .attr("fill", "transparent")
        .attr("stroke", "transparent")
    // rect over the short label. Will become opaque to cover this label
    // and replace with the long version on mouseover. 
    labelContainers
        .append("rect")
        .attr("class", "little-rect-cover")
        .attr("x", d => d == null ? undefined : -margin.left)
        .attr("y", (d: PlotDatum) => yaxisScale(d.member))
        .attr("width", 100)
        .attr("height", yaxisScale.bandwidth())
        .attr("fill", "transparent")
        .attr("stroke", "transparent")
    // y-axis labels
    labelContainers
        .append("text")
        .attr("class", "label-text")
        .attr("x", (d: PlotDatum) => d.member.length < 18 ? -5 : - margin.left)
        .attr("y", (d: PlotDatum) => yaxisScale((d.member)) + yaxisScale.bandwidth() / 1.5)
        .attr("text-anchor", (d: PlotDatum) => d.member.length < 18 ? "end" : "start")
        .attr("font-size", ".8em")
        .attr("fill", "transparent")
      .selectAll("tspan")
        .data((d: PlotDatum) => {
            let l = d.member
            const maxLength = Math.round( (config.
                width - margin.right - 10) / 5.5 )
            if (l.length > maxLength) {
                let splitIndex = l.indexOf(" ")
                while (splitIndex < maxLength) {
                    const next = l.indexOf(" ", splitIndex + 1)
                    if (next > -1 && next < maxLength) { splitIndex = next } else { break }
                }
                let newlabel = [l.slice(0, splitIndex), l.slice(splitIndex + 1)]
                if (newlabel[1].length > maxLength) newlabel[1] = newlabel[1].slice(0, maxLength - 4) + "..."
                return (newlabel)
            } else { return ([l]) }
        })
        .enter()
        .append("tspan")
        .attr("x", (d, i) => d.length < 18 && i === 0 ? -5 : - margin.left)
        .attr("dy", (d, i, data) => i > 0 ? "10" : data.length > 1 ? "-5" : "0")
        .text(d => d)
    
    

    // ---- update ----
    boxes
        .transition()
        .duration(500)
        .attr("width", function (d: PlotDatum) {
            return xaxisScale(d[countMetric]);
        })

    yaxisHoverLabels
        .selectAll(".label-number")
        .data(d => [d])
        .text(d => d[countMetric] == null ? 0 : d[countMetric])

    xaxisLabels.transition()
        .duration(500)
        .call(updateXaxis)
    xaxisGrid.transition()
        .duration(500)
        .call(updateXaxisGrid)
        .selectAll("line").attr("stroke", "#ddd")
    d3.selectAll(".domain").remove() // Removes the line on the bottom of the plot

    // ---- exit ----
    boxes.exit().remove();

    // ----- Mouseover event -----
    yaxisLongLabels.selectAll("g")
        .on("mouseover", function (d) {
            d3.select(this)
                .selectAll(".label-text")
                .transition()
                .duration(100)
                .attr("fill", "black")
                .attr("font-weight", "bold")
            d3.select(this)
                .selectAll(".label-number")
                .transition()
                .duration(100)
                .attr("fill", "black")
                .attr("font-weight", "bold")
            d3.select(this)
                .selectAll(".little-rect-cover")
                .transition()
                .duration(100)
                .attr("fill", "white")
        })
        .on("mouseout", function (d) {
            d3.select(this)
                .selectAll(".label-text")
                .transition()
                .duration(100)
                .attr("fill", "transparent")
                .attr("font-weight", "normal")
            d3.select(this)
                .selectAll(".label-number")
                .transition()
                .duration(100)
                .attr("fill", "transparent")
                .attr("font-weight", "normal")
            d3.select(this)
                .selectAll(".little-rect-cover")
                .transition()
                .duration(100)
                .attr("fill", "transparent")
        })

}

export const D3Barplot: D3Barplot = {
    create: createBarplot,
    update: updateBarplot
}
