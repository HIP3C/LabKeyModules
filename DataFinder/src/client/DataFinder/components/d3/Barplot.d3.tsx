import * as d3 from 'd3'
import { CubeDatum } from '../../../typings/CubeData';
import { entries } from 'd3';

// ================================================================== //
/* 
This is a barplot component which takes the following arguments in 
the props: 
  * data
  * name (id for the div to stick it in)
  * width
  * height

*/
// Types
interface DrawBarplotProps {
    data: CubeDatum[];
    name: string;
    width: number;
    height: number;
    labels: string[];
    countMetric: string;
    barColor: string;
}


export function drawBarplot(props: DrawBarplotProps) {

    const data = props.data;
    const name = props.name;

    const dataRange = [0, 10];
    const countMetric = props.countMetric
    data.forEach((v) => (v[countMetric] > dataRange[1]) && (dataRange[1] = v[countMetric]))

    const totalHeight = Math.max(195, 15 * props.labels.length + 20)

    const labels = props.labels.map(l => {
        let short = l
        if (l.length > 18) short = l.slice(0, 14) + "..."
        return ({ label: l, shortlabel: short })
    })
    // debuggers
    // const newLabels = [];

    // data.map((e, i) => {
    //     newLabels.push(e.label);
    //     // if (e.value > dataRange[1]) dataRange[1] = e.value;
    // });

    const svg = d3
        .select("#barplot-container-" + name).select("svg")
        .attr("height", totalHeight)
        .attr("width", 220)
        .attr("id", "barplot-container-" + name)

    const xAxisSvg = d3.select("#xaxis-" + name).select("svg")
        .attr("width", props.width)
        .attr("height", 40)


    // Create margins
    const margin = { top: 0, right: 15, bottom: 0, left: 100 },
        width = props.width - margin.left - margin.right,
        height = totalHeight - margin.top - margin.bottom;

    // Set scales using options

    const xaxisScale = d3
        .scaleLinear()
        .domain(dataRange).nice()
        .range([0, width]);

    const yaxisScale = d3
        .scaleBand()
        .domain(labels.map(l => l.label))
        .range([height, 0]);

    // Create body and axes
    // svg.append("g")
    //     .call(d3.axisLeft(yaxisScale));

    // svg.append("g")
    //     .call(d3.axisBottom(xaxisScale));

    let barplot: d3.Selection<SVGElement, any, HTMLElement, any>;
    let xaxisGrid, xaxisLabels, yaxisLabels
    if (!svg.selectAll("g").empty()) {
        svg.select("#xaxis-labels-" + name).remove()
        xaxisGrid = svg.select("#xaxis-grid-" + name)
        xaxisLabels = xAxisSvg.selectAll(".x.axis")
        barplot = svg.select("#barplot" + name)
    } else {
        barplot = svg
            .append("g")
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
            .attr("id", "barplot" + name)
            .attr("style", "overflow:auto")

        yaxisLabels = svg.append("g")
            .attr("id", "yaxis-labels-short-" + name)
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
        yaxisLabels.selectAll("text")
            .data(labels)
            .enter()
            .append("text")
            .attr("x", -5)
            .attr("y", (d) => {
                return yaxisScale((d.label)) + yaxisScale.bandwidth() / 1.5;
            })
            .attr("text-anchor", "end")
            .attr("font-size", ".8em")
            .text(d => d.shortlabel)

        const yaxisLongLabels = svg.append("g")
            .attr("id", "yaxis-labels-" + name)
            .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
        const labelContainers = yaxisLongLabels
            .selectAll("g")
            .data(data)
            .enter()
            .append("g")

        labelContainers
            .selectAll(".big-rect-cover")
            .data(d => [d])
            .enter()
            .append("rect")
            .attr("class", "big-rect-cover")
            .attr("x", - margin.left)
            .attr("y", d => yaxisScale(d.member))
            .attr("width", props.width)
            .attr("height", yaxisScale.bandwidth())
            .attr("fill", "transparent")
            .attr("stroke", "transparent")
        labelContainers
            .selectAll(".little-rect-cover")
            .data(d => [d])
            .enter()
            .append("rect")
            .attr("class", "little-rect-cover")
            .attr("x", d => d == null ? undefined : -margin.left)
            .attr("y", d => yaxisScale(d.member))
            .attr("width", 100)
            .attr("height", yaxisScale.bandwidth())
            .attr("fill", "transparent")
            .attr("stroke", "transparent")
        labelContainers.selectAll(".label-text")
            .data(d => [d])
            .enter()
            .append("text")
            .attr("class", "label-text")
            .attr("x", d => d.member.length < 18 ? -5 : - margin.left)
            .attr("y", d => yaxisScale((d.member)) + yaxisScale.bandwidth() / 1.5)
            .attr("text-anchor", d => d.member.length < 18 ? "end" : "start")
            .attr("font-size", ".8em")
            .attr("fill", "transparent")
            .text(d => d.member)
        labelContainers.selectAll(".label-number")
            .data(d => [d])
            .enter()
            .append("text")
            .attr("class", "label-number")
            .attr("x", props.width - margin.left - 30)
            .attr("y", d => yaxisScale((d.member)) + yaxisScale.bandwidth() / 1.5)
            .attr("text-anchor", "end")
            .text(d => d[countMetric] == null ? 0 : d[countMetric])
            .attr("font-size", ".8em")
            .attr("fill", "transparent")

        yaxisLongLabels.selectAll("g")
            .on("mouseover", function (d) {
                d3.select(this)
                    .selectAll(".label-text")
                    .transition()
                    .duration(100)
                    .attr("fill", "black")
                    .attr("font-weight", "bold")
                d3.select(this)
                    .selectAll(".label-number")
                    .transition()
                    .duration(100)
                    .attr("fill", "black")
                    .attr("font-weight", "bold")
                d3.select(this)
                    .selectAll(".little-rect-cover")
                    .transition()
                    .duration(100)
                    .attr("fill", "white")
            })
            .on("mouseout", function (d) {
                d3.select(this)
                    .selectAll(".label-text")
                    .transition()
                    .duration(100)
                    .attr("fill", "transparent")
                    .attr("font-weight", "normal")
                d3.select(this)
                    .selectAll(".label-number")
                    .transition()
                    .duration(100)
                    .attr("fill", "transparent")
                    .attr("font-weight", "normal")
                d3.select(this)
                    .selectAll(".little-rect-cover")
                    .transition()
                    .duration(100)
                    .attr("fill", "transparent")
            })

        xaxisGrid = svg.insert("g", ":first-child")
            .attr("id", "xaxis-grid-" + name)
            .attr(
                "transform",
                `translate(${margin.left}, ${(height + 5 + margin.top)})`
            )

        xaxisLabels = xAxisSvg.append("g")
            .attr("class", "x axis")
            .attr(
                "transform",
                `translate(${margin.left}, 0)`
            )

        xAxisSvg.append("g")
            .attr(
                "transform",
                `translate(${margin.left}, 0)`
            )
            .append("text")
            .attr("x", (width) / 2 )
            .attr("y", 30)
            .text(props.countMetric == "studyCount" ? "Studies" : "Participants")
            .attr("text-anchor", "middle")
            .attr("font-size", "0.8em")
    }

    xaxisGrid.transition()
        .duration(500)
        .call(d3.axisBottom(xaxisScale).ticks(3).tickSize(-(height + 5)).tickFormat(() => ""))
        .selectAll("line").attr("stroke", "#ddd")
    xaxisLabels.transition()
        .duration(500)
        .call(d3.axisBottom(xaxisScale).ticks(3).tickSize(0))
    // svg.append("g")
    //     .attr("id", "xaxis-labels-" + name)
    //     .attr(
    //         "transform",
    //         `translate(${margin.left}, ${(height + margin.top)})`
    //     )
    //     .call(d3.axisBottom(xaxisScale).tickValues([Math.round((dataRange[1] + 5) / 10) * 10]).tickSize(-height))
    //     .selectAll("line").attr("stroke", "#bcbcbc").attr("stroke-width", "3")

    d3.selectAll(".domain").remove()




    // add data
    const boxes = barplot.selectAll("rect").data(data);
    boxes
        .enter()
        .append("rect")
        .attr("class", "rect")
        .attr("x", xaxisScale(0))
        .attr("width", function (d: CubeDatum) {
            return xaxisScale(d[countMetric]);
        })
        .attr("y", function (d: CubeDatum) {
            return yaxisScale(d.member);
        })
        .attr("height", yaxisScale.bandwidth() - 1)
        .style("fill", props.barColor)
    boxes
        .transition()
        .duration(500)
        .attr("x", xaxisScale(0))
        .attr("width", function (d: CubeDatum) {
            return xaxisScale(d[countMetric]);
        })
        .attr("y", function (d: CubeDatum) {
            return yaxisScale(d.member);
        })
        .attr("height", yaxisScale.bandwidth() - 1)

    boxes.exit().remove();
}

