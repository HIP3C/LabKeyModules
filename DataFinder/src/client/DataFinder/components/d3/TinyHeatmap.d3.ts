import * as d3 from 'd3'
import { TinyHeatmapProps } from '../TinyHeatmap';
import { HeatmapDatum } from '../../../typings/CubeData'

// ================================================================== //

// Types



const drawTinyHeatmap = (props: TinyHeatmapProps) => {
    const heatmapInfo = props.heatmapInfo;
    const name = props.name;
    

    const svg = d3
        .select("#tinyheatmap-" + name)
        .attr("height", heatmapInfo.height)
        .attr("width", heatmapInfo.width)
        .attr("id", "tinyheatmap-" + name);

    // Create margins
    const margin = { top: 5, right: 0, bottom: 30, left: 50 },
        width = heatmapInfo.width - margin.left - margin.right,
        height = heatmapInfo.height - margin.top - margin.bottom;

    // Set scales using options
    const yaxisScale = heatmapInfo.yaxisScale
    const xaxisScale = heatmapInfo.xaxisScale
    const colorScale = heatmapInfo.colorScale


    // Create body and axes

    const heatmap = svg
        .append("g")
        .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
        .attr("id", "tinyheatmap" + name);
    const xaxislabels = svg
        .append("g")
        .attr("id", "xaxis-labels-" + name)
        .attr(
            "transform",
            "translate(" + margin.left + ", " + (height + margin.top) + ")"
        );
    // .call(d3.axisBottom(xaxisScale));

    const yaxislabels = svg
        .append("g")
        .attr("id", "yaxis-labels-" + name)
        .attr("transform", "translate(" + margin.left + "," + margin.top + ")")
        .call(d3.axisLeft(yaxisScale));

    xaxislabels.call(d3.axisBottom(xaxisScale))
        .selectAll("text")
        .attr("y", 5)
        .attr("x", -5)
        .attr("transform", "rotate(-45)")
        .style("text-anchor", "end")
        .attr("font-size", ".9em");

    // add data
    const boxes = heatmap.selectAll("rect").data(heatmapInfo.data);
    boxes
        .enter()
        .append("rect")
        .attr("x", (d: HeatmapDatum<any>) => {
            return xaxisScale(d.x);
        })
        .attr("width", xaxisScale.bandwidth() - 1)
        .attr("y", function (d: HeatmapDatum<any>) {
            return yaxisScale(d.y);
        })
        .attr("height", yaxisScale.bandwidth() - 1)
        .style("fill", function (d: HeatmapDatum<any>) {
            if (d.participantCount === 0 || d.y === undefined) return "transparent";
            return colorScale(d.participantCount);
        })
        .attr("stroke-width", "1px")
        .attr("stroke", "transparent");
}

export const D3TinyHeatmap = {drawTinyHeatmap}
