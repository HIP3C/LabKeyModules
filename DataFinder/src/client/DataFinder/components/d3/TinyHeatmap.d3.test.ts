import * as d3 from 'd3'
import {D3TinyHeatmap} from './TinyHeatmap.d3'
import { TinyHeatmapInfo } from '../../../typings/StudyCard'
import { createTinyHeatmapConsts, createTinyHeatmapYaxisScale } from '../TinyHeatmap'

describe("drawTinyHeatmap", () => {

    // setup
    const consts = createTinyHeatmapConsts()

    const info: TinyHeatmapInfo = {
        data: [
            {   x: "0",
                y: "HAI",
                participantCount: 15,
                studyCount: 1,
                data: undefined},
            {
                x: "1",
                y: "GE",
                participantCount: 30,
                studyCount: 1,
                data: undefined
            }
        ],
        assays: ["HAI", "GE"],
        height: 50,
        width: consts.width,
        yaxisScale: createTinyHeatmapYaxisScale(["HAI", "GE"]),
        xaxisScale: consts.xaxisScale,
        colorScale: consts.colorScale
    }

    document.body.innerHTML =
    `<div class="tinyheatmap" >
        <svg id="tinyheatmap-test"></svg>
    </div>`

    D3TinyHeatmap.drawTinyHeatmap({name: "test", heatmapInfo: info})
    const plot = document.getElementById('tinyheatmap-test')
    
    test("draw", () => {

        const plotTxt = plot.innerHTML
        expect(plotTxt).toMatchSnapshot()

    })

    test("yaxis", () => {
        const yaxis = document.getElementById("yaxis-labels-test")
        const labels = yaxis.getElementsByTagName("text")
        expect(labels.length).toEqual(2)
        expect(labels[1].innerHTML).toEqual("GE")
    })

    test("xaxis", () => {
        const xaxis = document.getElementById("xaxis-labels-test")
        const labels = xaxis.getElementsByTagName("text")
        expect(labels.length).toEqual(consts.timepoints.length)
    })

    test("rects", () => {
        const rects = document.getElementsByTagName("rect")
        expect(rects.length).toEqual(2)
    })




})