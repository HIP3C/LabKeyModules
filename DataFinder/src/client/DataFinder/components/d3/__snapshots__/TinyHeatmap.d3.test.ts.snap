// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawTinyHeatmap draw 1`] = `"<g transform=\\"translate(50,5)\\" id=\\"tinyheatmaptest\\"><rect x=\\"9.761904761904749\\" width=\\"8.761904761904763\\" y=\\"0\\" height=\\"9\\" style=\\"fill: #A1D99B;\\" stroke-width=\\"1px\\" stroke=\\"transparent\\"></rect><rect x=\\"19.52380952380951\\" width=\\"8.761904761904763\\" y=\\"10\\" height=\\"9\\" style=\\"fill: #74C476;\\" stroke-width=\\"1px\\" stroke=\\"transparent\\"></rect></g><g id=\\"xaxis-labels-test\\" transform=\\"translate(50, 20)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"middle\\"><path class=\\"domain\\" stroke=\\"currentColor\\" d=\\"M0.5,6V0.5H205.5V6\\"></path><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(4.880952380952367,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">&lt;0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(14.64285714285713,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">0</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(24.40476190476189,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">1</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(34.16666666666666,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">2</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(43.928571428571416,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">3</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(53.690476190476176,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">4</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(63.45238095238094,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">5</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(73.21428571428571,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">6</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(82.97619047619047,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">7</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(92.73809523809523,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">8</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(102.49999999999999,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">9</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(112.26190476190476,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">10</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(122.02380952380952,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">11</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(131.78571428571428,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">12</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(141.54761904761907,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">13</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(151.30952380952382,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">14</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(161.07142857142858,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">15-27</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(170.83333333333334,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">28</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(180.5952380952381,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">29-55</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(190.35714285714286,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">56</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(200.11904761904762,0)\\"><line stroke=\\"currentColor\\" y2=\\"6\\"></line><text fill=\\"currentColor\\" y=\\"5\\" dy=\\"0.71em\\" x=\\"-5\\" transform=\\"rotate(-45)\\" style=\\"text-anchor: end;\\" font-size=\\".9em\\">&gt;56</text></g></g><g id=\\"yaxis-labels-test\\" transform=\\"translate(50,5)\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\"><path class=\\"domain\\" stroke=\\"currentColor\\" d=\\"M-6,0.5H0.5V20.5H-6\\"></path><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,5)\\"><line stroke=\\"currentColor\\" x2=\\"-6\\"></line><text fill=\\"currentColor\\" x=\\"-9\\" dy=\\"0.32em\\">HAI</text></g><g class=\\"tick\\" opacity=\\"1\\" transform=\\"translate(0,15)\\"><line stroke=\\"currentColor\\" x2=\\"-6\\"></line><text fill=\\"currentColor\\" x=\\"-9\\" dy=\\"0.32em\\">GE</text></g></g>"`;
